time_test.go

```diff
@@ -68,3 +69,32 @@ func ExampleShortDuration() {
 	// 1d
 	// 1h0m1s
 }
+
+func BenchmarkShortDuration(b *testing.B) {
+	cases := []struct {
+		Name string
+		Data time.Duration
+	}{
+		{"Nanoseconds", time.Nanosecond * 123456},
+		{"Seconds", time.Second * 260},
+		{"Minutes", time.Minute * 59},
+		{"Hours", time.Hour * 5},
+		{"Days", time.Hour * 450},
```

Comment:
you should as few test cases as possible and keep them very different

we're not testing if the algorithm works for each time units, but how long it take for a simple case and for a complex case

this function has basically two extremes:

* super light -> 1*time.Second; because it will just have one unit, and that's done
* complex -> `time.Nanosecond * 1234567891234567891` which iterate over all checks and concatenate various units together (result = `14288d23h32m`)